crowbar_framework/app/models/api/node.rb

```diff
@@ -127,6 +127,31 @@ def upgrade
       true
     end
 
+    # Remove "pre-upgrade" attribute from given node
+    # We must do it from the node where pacemaker is running
+    def remove_pre_upgrade_role_for(name)
+      hostname = name.split(".").first
+      out = @node.run_ssh_cmd("crm node attribute #{hostname} set pre-upgrade false")
+      unless out[:exit_code].zero?
+        save_error_state("Channging the pre-upgrade role for #{hostname} from #{@node.name} failed")
+        return false
+      end
+      true
+    end
+
+    # Delete existing pacemaker resources, from other node in the cluster
```

Comment:
This code is not related to a particular node, so it doesn't make sense for it to be an instance method in the model.  Also, why is it in this model rather than in `Api::Upgrade`?